org.jacoco.core/src/org/jacoco/core/instr/Instrumenter.java

```diff
@@ -200,6 +203,12 @@ public int instrumentAll(final InputStream input,
 
 	private int instrumentZip(final InputStream input,
 			final OutputStream output, final String name) throws IOException {
+		if (generators.getLast() instanceof
+				OfflineInstrumentationCompanionAccessGenerator) {
+			generators
+					.addLast(new
+							OfflineInstrumentationCompanionAccessGenerator());
```

Comment:
@bjkail btw, I think that `Instrumenter` is non thread safe because of `setRemoveSignatures`